internal/terraform/node_resource_abstract_instance.go

```diff
@@ -802,29 +802,37 @@ func (n *NodeAbstractResourceInstance) plan(
 		priorVal = cty.NullVal(schema.ImpliedType())
 	}
 
-	log.Printf("[TRACE] Re-validating config for %q", n.Addr)
-	// Allow the provider to validate the final set of values.  The config was
-	// statically validated early on, but there may have been unknown values
-	// which the provider could not validate at the time.
-	//
-	// TODO: It would be more correct to validate the config after
-	// ignore_changes has been applied, but the current implementation cannot
-	// exclude computed-only attributes when given the `all` option.
-
-	// we must unmark and use the original config, since the ignore_changes
-	// handling below needs access to the marks.
-	unmarkedConfigVal, _ := origConfigVal.UnmarkDeep()
-	validateResp := provider.ValidateResourceConfig(
-		providers.ValidateResourceConfigRequest{
-			TypeName: n.Addr.Resource.Resource.Type,
-			Config:   unmarkedConfigVal,
-		},
-	)
-	diags = diags.Append(validateResp.Diagnostics.InConfigBody(config.Config, n.Addr.String()))
-	if diags.HasErrors() {
-		return plan, state, keyData, diags
-	}
+	// Config generated from a resource's import state may fail validation in
+	// the case of schema behaviours such as ExactlyOneOf and ConflictsWith.
+	// We don't want to fail the plan now, because that would give the user no
+	// way to proceed and fix the config to make it valid. We allow the plan to
+	// complete and output the generated config.
+	// TODO: Run Validate and output the errors as warnings?
```

Comment:
If this errors the entire plan exits with error, e.g. 
```hcl
❤ @katy ➜  1  terraform plan
╷
│ Error: Invalid or unknown key
│ 
│   with aws_instance.ubuntu,
│   on g.tf line 17, in resource "aws_instance" "ubuntu":
│   17:   id                                   = "i-0f486fb0b37793c7d"
│ 
╵
╷
│ Error: Conflicting configuration arguments
│ 
│   with aws_instance.ubuntu,
│   on g.tf line 20, in resource "aws_instance" "ubuntu":
│   20:   ipv6_address_count                   = 0
│ 
│ "ipv6_address_count": conflicts with ipv6_addresses
╵
╷
│ Error: Conflicting configuration arguments
│ 
│   with aws_instance.ubuntu,
│   on g.tf line 21, in resource "aws_instance" "ubuntu":
│   21:   ipv6_addresses                       = []
│ 
│ "ipv6_addresses": conflicts with ipv6_address_count
```

So no plan or generated config is produced.

>If the configuration would generate validation errors, applying that configuration is only hiding the issue from the user, and passing on to a later workflow step which may not be expecting invalid configuration.

Yes. The later workflow step is the only step in which the user is able to resolve the validation errors. It's not possible to resolve them by any change to config before the plan, because they are due to the provider returning invalid state, which the user can't prevent.

Reply:
Perhaps the best option here is to return the validate diags, but when generating config the errors should be converted to warnings.

This allows the plan to succeed and the user to see the validate errors without having to run a further `terraform validate` or `terraform plan`. They can then fix the errors in the generated config.

Reply:
Fortunately, such errors are now a lot less common with the fixes in this PR.

In particular none of the above validation errors occur when generating config for a typical `aws_instance`.